lib/pure/asyncnet.nim

```diff
@@ -713,25 +713,27 @@ elif defined(nimdoc):
 
 proc close*(socket: AsyncSocket) =
   ## Closes the socket.
-  defer:
-    socket.fd.AsyncFD.closeSocket()
-  when defineSsl:
-    if socket.isSsl:
-      let res =
-        # Don't call SSL_shutdown if the connection has not been fully
-        # established, see:
-        # https://github.com/openssl/openssl/issues/710#issuecomment-253897666
-        if not socket.sslNoShutdown and SSL_in_init(socket.sslHandle) == 0:
-          ErrClearError()
-          SSL_shutdown(socket.sslHandle)
-        else:
-          0
-      SSL_free(socket.sslHandle)
-      if res == 0:
-        discard
-      elif res != 1:
-        raiseSSLError()
-  socket.closed = true # TODO: Add extra debugging checks for this.
+  if not socket.closed:
```

Comment:
Prefer: 
```
if socket.closed: return
```

Style rather than wrapping everything in more and more nested ifs.

Reply:
No, nested ifs are better.